src/Microsoft.Health.Fhir.Shared.Api/Controllers/ValidateController.cs

```diff
@@ -39,23 +42,58 @@ public ValidateController(IMediator mediator)
         [HttpPost]
         [Route(KnownRoutes.ValidateResourceType)]
         [AuditEventType(AuditEventSubType.Read)]
-        public async Task<IActionResult> Validate([FromBody] Resource resource)
+        public async Task<IActionResult> Validate([FromBody] Resource resource, [FromQuery(Name = "profile")] Uri profile)
         {
-            return await RunValidationAsync(resource);
+            if (resource.ResourceType == ResourceType.Parameters)
+            {
+                var parameterResource = (Parameters)resource;
+                var profileFromParameters = parameterResource.Parameter.Find(param => param.Name.Equals("profile", StringComparison.OrdinalIgnoreCase));
+                if (profileFromParameters != null)
+                {
+                    if (profile != null)
+                    {
+                        throw new BadRequestException(Api.Resources.MultipleProfilesProvided);
+                    }
+
+                    if (profileFromParameters.Value == null)
+                    {
+                        throw new BadRequestException("Profile parameter is empty");
+                    }
+
+                    try
+                    {
+                        profile = new Uri(profileFromParameters.Value.ToString());
+                    }
+                    catch
+                    {
+                        throw new BadRequestException("Profile is invalid.");
+                    }
+                }
+
+                resource = parameterResource.Parameter.Find(param => param.Name.Equals("resource", StringComparison.OrdinalIgnoreCase)).Resource;
+            }
```

Comment:
I'd suggest moving this code to [ValidateResourceOperationValidator.cs](https://github.com/microsoft/fhir-server/blob/main/src/Microsoft.Health.Fhir.Core/Features/Operations/Validate/ValidateResourceOperationValidator.cs) or removing that class.

I was looking at this approach also: https://github.com/microsoft/fhir-server/blob/30d15e0b5cfafa34026394d3218c4d12ce8dd75a/src/Microsoft.Health.Fhir.CosmosDb/Features/Operations/SortSettings/SortParametersValidator.cs

Reply:
Not sure I get it, let me ping you later....

---
In reply to: [574185717](https://github.com/microsoft/fhir-server/pull/1654#discussion_r574185717) [](ancestors = 574185717)